tests/test_diffraction_StaticStructureFactor.py

```diff
@@ -0,0 +1,394 @@
+                Minimum :math:`k` value to include in the calculation.
+        """
+        system = freud.locality.NeighborQuery.from_system(system)
+        N = len(system.points)
+
+        Q = np.linspace(k_min, k_max, bins, endpoint=False)
+        Q += (k_max - k_min) / bins / 2
+        S = np.zeros_like(Q)
+
+        # Compute all pairwise distances
+        distances = system.box.compute_all_distances(
+            system.points, system.points
+        ).flatten()
+
+        for i, q in enumerate(Q):
+            S[i] += np.sum(np.sinc(q * distances / np.pi)) / N
+
+        return Q, S
+
+    def test_debye_validation(self):
+        """Validate the Debye method against a Python implementation."""
+        bins = 1000
+        k_max = 100
+        k_min = 0
+        sf = freud.diffraction.StaticStructureFactorDebye(bins, k_max, k_min)
+        box, points = freud.data.UnitCell.fcc().generate_system(4, sigma_noise=0.01)
+        system = freud.locality.NeighborQuery.from_system((box, points))
+        sf.compute(system)
+        Q, S = self._validate_debye_method(system, bins, k_max, k_min)
+        npt.assert_allclose(sf.bin_centers, Q)
+        npt.assert_allclose(sf.S_k, S, rtol=1e-5, atol=1e-5)
+
+    def test_debye_ase(self):
+        """Validate Debye method agains ASE implementation"""
+        ase = pytest.importorskip("ase")
+        asexrd = pytest.importorskip("ase.utils.xrdebye")
+        bins = 1000
+        k_max = 100
+        k_min = 0
+        box, points = freud.data.UnitCell.fcc().generate_system(4, sigma_noise=0.01)
+        # ase implementation has no PBC taken into account
+        box.periodic = False
+        system = freud.locality.NeighborQuery.from_system((box, points))
+        sf = freud.diffraction.StaticStructureFactorDebye(bins, k_max, k_min)
+        sf.compute(system)
+        # ASE system generation as atoms object
+        atoms = ase.Atoms(
+            positions=points, pbc=True, cell=box.L, numbers=np.ones(len(points))
+        )
+        xrd = asexrd.XrDebye(
+            atoms=atoms, wavelength=1.0, method=None, damping=0.0, alpha=1.0
+        )
+        # calculate S_k for given set of k values
+        S_ase = xrd.calc_pattern(sf.bin_centers, mode="SAXS") / len(points)
+        npt.assert_allclose(sf.S_k, S_ase, rtol=1e-5, atol=1e-5)
+
+
+class TestStaticStructureFactorDirect(StaticStructureFactorTest):
+    @classmethod
+    def build_structure_factor_object(
+        cls, bins, k_max, k_min=0, num_sampled_k_points=0
+    ):
+        return freud.diffraction.StaticStructureFactorDirect(
+            bins, k_max, k_min, num_sampled_k_points
+        )
+
+    @classmethod
+    def get_large_k_params(cls):
+        return {"bins": 100, "k_max": 500, "k_min": 400, "num_sampled_k_points": 200000}
+
+    @classmethod
+    def get_min_valid_k(cls, Lx, Ly, Lz=None):
+        if Lz == None:
+            min_length = np.min([Lx, Ly])
+        else:
+            min_length = np.min([Lx, Ly, Lz])
```

Comment:
> Comparisons to singletons like None should always be done with is or is not, never the equality operators. ([PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations))
```suggestion
        min_length = np.min([Lx, Ly]) if Lz is None else np.min([Lx, Ly, Lz])
```

Same suggestion applies above.